src/odb_loose.c

```diff
@@ -152,6 +153,11 @@ static int parse_header(
 		size < 0)
 		goto on_error;
 
+	if ((uint64_t)size > SIZE_MAX) {
+		giterr_set(GITERR_OBJECT, "object is larger than available memory");
```

Comment:
This is technically not correct, as this is not about available memory (which we don't know about right now). Rather, this should be somethink like "object size exceeds address space"?

Reply:
Well, I think _that's_ not _technically_ correct either.  `size_t` doesn't reflect the address space, it reflects the size of the largest single object size.  But I think that this is _practically_ correct for the set of computers that are actually in use, and that this is a much more actionable error message.

Reply:
Yeah, guess one should be a bit more precise when nagging about technical incorrectness :P